src/OpenTelemetry.Collector.AspNetCore/RequestsCollector.cs

```diff
@@ -33,16 +34,17 @@ public class RequestsCollector : IDisposable
         /// Initializes a new instance of the <see cref="RequestsCollector"/> class.
         /// </summary>
         /// <param name="options">Configuration options for dependencies collector.</param>
-        /// <param name="tracer">Tracer to record traced with.</param>
+        /// <param name="tracerFactory">TracerFactory which creates the Tracer to record traced with.</param>
         /// <param name="sampler">Sampler to use to sample dependency calls.</param>
-        public RequestsCollector(RequestsCollectorOptions options, ITracer tracer, ISampler sampler)
+        public RequestsCollector(RequestsCollectorOptions options, ITracerFactory tracerFactory, ISampler sampler)
         {
+            const string name = "Microsoft.AspNetCore";
             this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber<HttpRequest>(
-                new Dictionary<string, Func<ITracer, Func<HttpRequest, ISampler>, ListenerHandler<HttpRequest>>>()
+                new Dictionary<string, Func<ITracerFactory, Func<HttpRequest, ISampler>, ListenerHandler<HttpRequest>>>()
                 {
-                    { "Microsoft.AspNetCore", (t, s) => new HttpInListener(t, s) },
+                    { name, (t, s) => new HttpInListener(name, Assembly.GetExecutingAssembly().GetName().Version, t, s) },
```

Comment:
this should create a tracer from factory specif to "Microsoft.AspNetCore" and provide a version of ASP.NET Core rather than executing assembly

Reply:
Actually it should be the other way around:
Executing assembly is right since we want to get version and name of the "instrumentation library" (instead of the instrumented library).
My mistake was to use *Microsoft.AspnetCore*. This should rather be *OpenTelemetry.Collector.AspNetCore* here.

Reply:
Assembly.GetExecutingAssembly() is user application, rather than opentelemetry assembly

Reply:
You're right, of course.
However, that code has changed in the meantime and the assembly version is fetched using type information now.